velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/substrait/VeloxToSubstraitExpr.h"
+
+#include "velox/expression/ControlExpr.h"
+
+namespace facebook::velox::substrait {
+
+void VeloxToSubstraitExprConvertor::transformVeloxExpr(
```

Comment:
This method is very large. Consider extracting helper method to convert each type of expression. Also, consider converting a small set of expressions in the PR and add more expression types along with tests in a follow-up PR.